taichi/ir/expr.cpp

```diff
@@ -127,17 +127,17 @@ Expr Expr::eval() const {
 
 void Expr::operator+=(const Expr &o) {
   if (this->atomic) {
-    current_ast_builder().insert(Stmt::make<FrontendAtomicStmt>(
-        AtomicOpType::add, ptr_if_global(*this), load_if_ptr(o)));
+    (*this) = Expr::make<AtomicOpExpression>(
+        AtomicOpType::add, ptr_if_global(*this), load_if_ptr(o));
   } else {
     (*this) = (*this) + o;
   }
 }
 
 void Expr::operator-=(const Expr &o) {
   if (this->atomic) {
-    current_ast_builder().insert(Stmt::make<FrontendAtomicStmt>(
-        AtomicOpType::add, *this, -load_if_ptr(o)));
+    (*this) = Expr::make<AtomicOpExpression>(
+        AtomicOpType::add, ptr_if_global(*this), -load_if_ptr(o));
```

Comment:
Why not
```suggestion
    (*this) = Expr::make<AtomicOpExpression>(
        AtomicOpType::sub, ptr_if_global(*this), load_if_ptr(o));
```
? Since you did the add->sub translation below.

Reply:
Again, there's nothing changed, please compare with the old code.

Reply:
That said, I think it's better to do the conversion once in `flatten`, taking the suggestion, thanks!